src/tricks.rs

```diff
@@ -45,18 +64,55 @@ impl Trick {
         let mut report = TrickReport::new(&self.name);
         report.set_system_info();
 
+        let mut create_vm = 0;
+
         for step in &self.steps {
-            status = step.run().await;
+            if create_vm == 0 {
+                if let steps::Step::createEnvironment( step ) = step {
+                    //run create environment
+                    //rest of steps get sent to VM
+                    if !is_vm {
+                        create_vm = 1;
+                    }
+                    else {
+                        continue;
+                    }
+                }
 
-            if let Step::SpawnContainer(step) = step {
-                containers.insert(step.name.to_owned());
-            }
+                if create_vm == 1 {
+                    status = step.run().await;
+                    let mut client = api::client::HoudiniVsockClient::new(3, 2375).await.unwrap();
+                    let step_report = StepReport::new(step, status);
+                    report.add(step_report);
 
-            let step_report = StepReport::new(step, status);
-            report.add(step_report);
+                    let step_report = client.trick(serde_json::to_vec(self).unwrap()).await.unwrap();
 
-            if status.is_final() {
-                break;
+                    for step in step_report.steps {
+                        report.add(step);
+                    }
+
+                    report.set_status(step_report.status);
+
+                    
+
+                    break;
+                }
+
+                status = step.run().await;
+
+                if let Step::SpawnContainer(step) = step {
+                    containers.insert(step.name.to_owned());
+                }
+
+                let step_report = StepReport::new(step, status);
+                report.add(step_report);
```

Comment:
Not really sure what's going on here. Why are we creating a new step report for every step and copying all steps over? That's really slow and I don't think it's necessary, unless I'm missing something?